app/src/main/java/protect/card_locker/CatimaRemoteViewsService.kt

```diff
@@ -0,0 +1,132 @@
+package protect.card_locker
+
+import android.content.Context
+import android.content.Intent
+import android.content.SharedPreferences
+import android.database.sqlite.SQLiteDatabase
+import android.graphics.Color
+import android.graphics.drawable.Icon
+import android.os.Build
+import android.util.Log
+import android.view.View
+import android.widget.RemoteViews
+import android.widget.RemoteViewsService
+import androidx.annotation.RequiresApi
+import protect.card_locker.DBHelper.LoyaltyCardArchiveFilter
+
+
+class CatimaWidgetRemoteViewsFactory(private var context: Context) :
+    RemoteViewsService.RemoteViewsFactory {
+
+    private var mDatabase: SQLiteDatabase = DBHelper(context).readableDatabase
+
+    override fun onCreate() {
+        onDataSetChanged()
+    }
+
+    private var mCards: ArrayList<LoyaltyCard> = ArrayList<LoyaltyCard>()
+    private var mOrderDirection = DBHelper.LoyaltyCardOrderDirection.Ascending
+    private var mOrder =  DBHelper.LoyaltyCardOrder.LastUsed
+
+    override fun onDataSetChanged() {
+        val sortPref: SharedPreferences = context.getSharedPreferences(
+            "sharedpreference_sort",
+            Context.MODE_PRIVATE
+        )
+
+        val orderString = sortPref.getString("sharedpreference_sort_order", null)
+        val orderDirectionString =
+            sortPref.getString("sharedpreference_sort_direction", null)
+
+        if (orderString != null && orderDirectionString != null) {
+            try {
+                mOrder = DBHelper.LoyaltyCardOrder.valueOf(orderString)
+                mOrderDirection = DBHelper.LoyaltyCardOrderDirection.valueOf(orderDirectionString)
+            } catch (ignored: IllegalArgumentException) {
+            }
+        }
+
+        val cur = DBHelper.getLoyaltyCardCursor(
+            mDatabase,
+            "",
+            null,
+            mOrder,
+            mOrderDirection,
+            LoyaltyCardArchiveFilter.All
+        )
```

Comment:
While I agree people should be able to configure the widget as they want, the widget changing based on what you set in Catima's home screen seems confusing to me.

I see a lot of widgets ask you how you want to configure them when you place them, but that is probably overkill for version 1 :)

For version 1, maybe we should just show the "default": alphabetical (and, to prevent it from being too much, only the Unarchived cards).

Reply:
Really? It seems confusing to me for a widget's "settings" to change based on stuff changed outside of the widget. But if you say so, sure I guess.

We should probably at least default to `LoyaltyCardArchiveFilter.Unarchived`, to not include archived cards, especially because your version doesn't show the archive icon on archived cards.